symplyphysics/laws/electricity/circuits/resistance_of_the_serial_RLC_circuit.py

```diff
@@ -0,0 +1,53 @@
+from sympy import (Eq, solve, sqrt,)
+from symplyphysics import (units, Quantity, Symbol, Vector, print_expression, validate_input,
+    validate_output, vector_magnitude,)
+from symplyphysics.core.expr_comparisons import expr_equals
+
+# Description
+## Consider an electrical circuit consisting of a capacitor, coil, and resistor connected in series.
+## Then you can find the total resistance of such a circuit.
+
+## Law is: R = sqrt(R0^2 + (Xl - Xc)^2), where
+## R - resistance of circuit,
+## R0 - resistance of resistor
+## Xl - inductive resistance,
+## Xc - capacitive resistance.
+
+circuit_resistance = Symbol("circuit_resistance", units.impedance)
+
+resistance_resistor = Symbol("resistance_resistor", units.impedance)
+capacitive_resistance = Symbol("capacitive_resistance", units.impedance)
+inductive_resistance = Symbol("inductive_resistance", units.impedance)
+
+law = Eq(circuit_resistance, sqrt(resistance_resistor**2 + (inductive_resistance - capacitive_resistance)**2))
+
+# This law might be derived via impedance definition. The impedance is Z = R0 + j * X.
+# Then the resistance of the circuit is equal to the impedance modulus and is equal to sqrt(R0^2 + X^2).
+# X = Xc - Xl
+# Then, as a result, the resistance of the circuit is equal to sqrt(R0^2 + (Xl - Xc)^2).
+
+# The complex impedance can be represented as a vector with two components: the real and imaginary parts.
+# Let's find the modulus of this vector and get the resistance of the circuit.
+
+resistance_vector = Vector([resistance_resistor, inductive_resistance - capacitive_resistance])
+resistance_derived = vector_magnitude(resistance_vector)
```

Comment:
You can take `abs` of complex number to get the same result. No need to use vectors here.

Reply:
> You can take `abs` of complex number to get the same result. No need to use vectors here.

Yes, I tried to do that. But the resulting expression "circuit_resistance_derived" was not equal to my law.
The code of this attempt is below.

from symplyphysics.definitions import impedance_is_resistance_and_reactance as impedance_law

impedance_law_applied = impedance_law.definition.subs({
impedance_law.resistance: resistance_resistor,
impedance_law.reactance: inductive_resistance - capacitive_resistance,
})
circuit_resistance_derived = solve(impedance_law_applied, impedance_law.impedance, dict=True)[0][impedance_law.impedance]

# Check if derived resistance is same as declared.
assert expr_equals(abs(circuit_resistance_derived), law.rhs)

Reply:
It is because symbols are complex by default. Add `real=True` to symbol definitions, eg `inductive_resistance = Symbol("inductive_resistance", units.impedance, real=True)` to make `abs()` work.